packages/renderer/src/lib/preferences/PreferencesConnectionCreationOrEditRendering.svelte

```diff
@@ -217,6 +237,29 @@ async function loadConnectionParams(): Promise<void> {
   }
 }
 
+function buildAuditData(): AuditRequestItems {
+  const data: { [key: string]: unknown } = {};
+
+  if (existingFormData) {
+    Object.assign(data, existingFormData);
+    return data as AuditRequestItems;
+  }
+
+  for (const field of configurationKeys) {
+    if (field.id) {
+      const value = configurationValues.get(field.id);
+      data[field.id] = value?.value ?? field.default;
+    }
+  }
+
+  return data as AuditRequestItems;
+}
```

Comment:
suggestion: remove casts

```suggestion
function buildAuditData(): AuditRequestItems {
  const data: AuditRequestItems = {};

  if (existingFormData) {
    Object.assign(data, existingFormData);
    return data;
  }

  for (const field of configurationKeys) {
    if (field.id) {
      const value = configurationValues.get(field.id);
      data[field.id] = value?.value ?? field.default;
    }
  }

  return data;
}
```